zulipterminal/ui_tools/boxes.py

```diff
@@ -80,6 +86,28 @@ def stream_box_view(self, button: Any=None, caption: str='',
         ]
         self.contents = write_box
 
+    def autocomplete(self, text: Any, state: int) -> Any:
+        if text.startswith('@'):
+            users = self.view.users.copy()
+            users_list = users.copy()
+            for user in users:
+                if not match_user(user, text[1:]):
+                    users_list.remove(user)
```

Comment:
Can you combine this with the loop below? It seems confusing to have two lists here - was there an alternative implementation you were looking at that used this, or a future version which would?

Reply:
the first loop is used to remove all the not-matching users from users_list, second loop creates a new list that is iterated by user for the suggestions.

Reply:
@omi10859, I believe you could do that without two loops too.
```python3
tmp = []
for user in users:
    if match_user(user, text[1:]):
        tmp.append('@**%s**' % user['full_name'])
```

Reply:
Can you suggest me a way?